src/main/java/io/openliberty/tools/intellij/lsp4mp/lsp4ij/LanguageServerWrapper.java

```diff
@@ -636,6 +629,12 @@ private CompletableFuture<LanguageServer> connect(@Nonnull URI absolutePath, Doc
                 }
                 DocumentContentSynchronizer listener = new DocumentContentSynchronizer(this, theDocument, syncKind);
                 theDocument.addDocumentListener(listener);
+                Set<DocumentContentSynchronizer> synchronizers = theDocument.getUserData(DocumentContentSynchronizer.KEY);
+                if (synchronizers == null) {
+                    synchronizers = Collections.synchronizedSet(new LinkedHashSet<>());
+                    theDocument.putUserData(DocumentContentSynchronizer.KEY, synchronizers);
+                }
+                synchronizers.add(listener);
```

Comment:
Is it going to mess anything up if a listener is added as `theDocument.addDocumentListener(listener);` and to the set `synchronizers.add(listener)` ?

Reply:
I'm not sure what you're suggesting could happen. The set is custom user data. The only code that reads or writes it is in this PR. It adds the `DocumentContentSynchronizer`s to the set that it expects to access later if the config source cache is cleared.

Reply:
I read the comment in that file and now I understand.